### ISSUE TITLE
errors lost

### ISSUE THREAD

**Issue by rosskevin** [final-form/final-form#75]
### Are you submitting a **bug report** or a **feature request**?
bug?

### What is the current behavior?
Existing validations are lost on form change.  

This _may_ be related to the change in #73.

![screen recording 2018-01-31 at 10 59 54 am](https://user-images.githubusercontent.com/136564/35636410-a3ee1cfe-0676-11e8-9715-4594f6a36dc0.gif)

Note that I have tried this with the async field validation example (with updated dependencies) and **it works**, so something I'm doing differs from it.  I do not yet know what I am doing differently to cause this.

### What is the expected behavior?
validations are not lost

### Sandbox Link
Working example with latest dependencies https://codesandbox.io/s/wkvl93qyx5

### What's your environment?
```
yarn list --pattern final-form
yarn list v1.3.2
├─ final-form-arrays@1.0.4
├─ final-form@4.2.0
├─ react-final-form-arrays@1.0.3
└─ react-final-form@3.1.0
```

## TL;DR
I could not reproduce with example, but it was not doing this yesterday before #73.  If you have any guidance as to the potential source of the difference, I'd like to try to boil this down to a reproducible test case.  So far, I have had no luck.

Any thoughts  @erikras? I'll get back to this after lunch.

**Comment by erikras** [comment]
I'd need to see more code, I think.

**Comment by rosskevin** [comment]
I'm still on this and I'll update it with my findings - but it is clearly not reproducible so I'm going to close it.

**Comment by rosskevin** [comment]
Got it!  Wow, that took me quite a while to parse through my changes in my new implementation over the past few days.

Here is the reproduction starting from the async field validations sample and adding `validateFields={[]}` for optimizing/isolating rendering per field:

https://codesandbox.io/s/wkvl93qyx5

**Comment by madhums** [comment]
I am facing the same issue! Suppose there are two fields, name and username, and username has error, on changing of name field, the error on username field disappears. This can be reproduced in https://codesandbox.io/s/wkvl93qyx5 as mentioned above.

**Comment by erikras** [comment]
@ek5000 Great sandbox example. 👍 Will investigate... 🔍

**Comment by mahulst** [comment]
@erikras Any updates? Can I help you with the investigation? 
Or if you are busy do have any pointers where to start the investigation?

**Comment by erikras** [comment]
Good news. I've created a failing test. Expect resolution soon...

**Comment by erikras** [comment]
Published fix in [`v4.4.0`](https://github.com/final-form/final-form/releases/tag/v4.4.0).
